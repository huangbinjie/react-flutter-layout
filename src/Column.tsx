import React from 'react'
import { Overflow } from './style'

export interface IColumn {
  className?: string
  overflow?: Overflow
}

/**
 * 上下布局的组件。
 * 高宽默认占满父容器。
 * 默认占满父容器
 */
export default function Column(props: React.PropsWithChildren<IColumn>) {
  const {
    children,
    overflow = 'hidden',
    ...restProps
  } = props

  const style: React.CSSProperties = {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    overflow
  }

  return <div style={style} {...restProps}>{children}</div>
}
