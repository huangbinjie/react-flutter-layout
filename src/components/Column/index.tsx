import React from 'react'
import { style as jss } from 'typestyle'
import classnames from 'classnames'
import { MainAxisSize, DirectionContext, Alignment, decodeAlignment } from '../../style'

export type ColumnProps = {
  className?: string
  mainAxisAlignment?: Alignment
  crossAxisAlignment?: Alignment
  mainAxisSize?: MainAxisSize
  style?: React.CSSProperties
}

/**
 * 上下布局的组件。
 * 高宽尽可能的大。
 */
export default function Column({ className = '', crossAxisAlignment = 'normal', mainAxisAlignment = 'start', mainAxisSize = 'max', children, ...restProps }: React.PropsWithChildren<ColumnProps>) {
  const classname = jss({
    display: 'flex',
    flexDirection: 'column',
    alignItems: decodeAlignment(crossAxisAlignment),
    justifyContent: decodeAlignment(mainAxisAlignment),
    width: "100%",
    height: decodeMainAxisSize(mainAxisSize)
  })

  return (
    <DirectionContext.Provider value="column">
      <div debug-label="Column" className={classnames(classname, className)} {...restProps}>{children}</div>
    </DirectionContext.Provider>
  )
}

export function decodeMainAxisSize(size: MainAxisSize) {
  if (size === 'min') {
    return 'auto'
  }

  return '100%'
}
