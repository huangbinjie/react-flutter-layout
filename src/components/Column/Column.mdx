---
name: Column
menu: components
---

import { Playground, Props } from 'docz'
import Container from '../Container'
import Column from '../Column'
import Center from '../Center'

## Column

提供垂直布局的组件。等价于 `flex-direction: column`

### Basic

<Playground>
  <Column>
    <Container width="100px" height="100px" color="#ececec">
      <Center>居中</Center>
    </Container>
    <Container width="100px" height="100px" color="blue">
      <Center style={{color: "#fff"}}>居中</Center>
    </Container>
  </Column>
</Playground>

## Props

<Props of={Column} />
