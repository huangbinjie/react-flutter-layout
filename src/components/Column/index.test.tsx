import React from 'react'
import renderer from 'react-test-renderer'

import Column from '.'

describe('Column', () => {
  test('basic', () => {
    const tree = renderer.create(
      <Column>empty content</Column>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
