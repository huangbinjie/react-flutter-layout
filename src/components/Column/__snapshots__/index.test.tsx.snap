// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column basic 1`] = `
<div
  className="ftch486"
  debug-label="Column"
>
  empty content
</div>
`;
